app/api/testing/firebase-test-utils.ts

```
@@ -0,0 +1,147 @@
+import { FirebaseApiClient } from '../FirebaseApiClient';
+import { ApiInterface, UserInfo } from '../ApiInterface';
+
+/**
+ * Mock FirebaseApiClient for testing
+ * Implements the same interface but allows mocking of all methods
+ */
+export class MockFirebaseApiClient implements ApiInterface {
+  private connected: boolean = false;
+  private mockUser: UserInfo | null = null;
+
+  constructor() {}
```

Biome (1.9.4)

[error] This constructor is unnecessary.

Unsafe fix: Remove the unnecessary constructor.


(lint/complexity/noUselessConstructor)